From: Nico Grunbaum <na-g@nostrum.com>
Date: Thu, 23 May 2024 17:28:00 +0000
Subject: Bug 1897459 - P0 - for BSD do not build physical_socket_server in
 libwebrtc;r=mjf,webrtc-reviewers

libwebrtc added the use of several non-portable socket options in this code.
Firefox does not use any of the code in this file, but there is a header dependency chain leading from code that is used.

Differential Revision: https://phabricator.services.mozilla.com/D210858
Mercurial Revision: https://hg.mozilla.org/mozilla-central/rev/4e7b3c485e549e7e35b227b517acadbbad957004
---
 rtc_base/BUILD.gn | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/rtc_base/BUILD.gn b/rtc_base/BUILD.gn
index af8726c77c..39348672f6 100644
--- a/rtc_base/BUILD.gn
+++ b/rtc_base/BUILD.gn
@@ -1033,6 +1033,12 @@ rtc_library("threading") {
     "thread.cc",
     "thread.h",
   ]
+  if (is_bsd && build_with_mozilla) {
+    sources -= [
+      "physical_socket_server.cc",
+      "physical_socket_server.h",
+    ]
+  }
   absl_deps = [
     "//third_party/abseil-cpp/absl/algorithm:container",
     "//third_party/abseil-cpp/absl/base:core_headers",
